### DISCUSSION TITLE
Configure Babel to transpile node modules

### DISCUSSION THREAD

**Discussion by qiulang** [babel/babel#13507] [Q&A]
I raised this issue [
@babel/preset-env fails to polyfill IE11 for debug module, version 4 ](https://github.com/babel/babel/issues/10707) in 2019. It was closed by @nicolo-ribaudo with the words "If you configure Babel to transpile node modules but it still isn't working, please provide a minimal repository I can clone to investigate."

During that time IE support is not a must so I didn't further interact with him. But unfortunately now my customers need me support IE 11 so I am working on it again 😂

The issue was never resolved there with the demo project I created https://github.com/qiulang/IE11

In my issue  @nicolo-ribaudo said "this is not a debug bug, but a problem with how Babel has been configured." 

Can someone help me out here ? This is my setting https://github.com/qiulang/IE11/blob/master/babel.config.js

Thanks a lot.

Qiulang

**Comment by 2239559319** [comment]
> I raised this issue [ @babel/preset-env fails to polyfill IE11 for debug module, version 4 ](https://github.com/babel/babel/issues/10707) in 2019. It was closed by @nicolo-ribaudo with the words "If you configure Babel to transpile node modules but it still isn't working, please provide a minimal repository I can clone to investigate."
> 
> During that time IE support is not a must so I didn't further interact with him. But unfortunately now my customers need me support IE 11 so I am working on it again 😂
> 
> The issue was never resolved there with the demo project I created https://github.com/qiulang/IE11
> 
> In my issue @nicolo-ribaudo said "this is not a debug bug, but a problem with how Babel has been configured."
> 
> Can someone help me out here ? This is my setting https://github.com/qiulang/IE11/blob/master/babel.config.js
> 
> Thanks a lot.
> 
> Qiulang

不能在ie11上跑的原因是因为debug这个依赖并没有编译成es5的代码，是直接引入的es6的代码，vue的打包默认是不会去打包node_modules里面的文件，这种情况在vue.config.js里面设置transpileDependencies就好了。
--------
最好别在vue脚手架起的项目里面改babel的配置
以下都是用的默认vue脚手架生成的配置
这是默认打包出的文件，公共依赖里面的debug没有被编译
![screenshot-20210711-000523](https://user-images.githubusercontent.com/34960995/125169651-c986aa80-e1dd-11eb-869d-32236fa78bb9.png)
--------
这是设置transpileDependencies后的打包文件，应该是可以在ie11上运行的(代码都是es5)（我的电脑是mac所以没测试）
![screenshot-20210711-001527](https://user-images.githubusercontent.com/34960995/125169704-fc30a300-e1dd-11eb-9622-3f71432bb6a7.png)
![screenshot-20210711-001542](https://user-images.githubusercontent.com/34960995/125169769-43b72f00-e1de-11eb-9c61-8627986d067a.png)

**Reply by qiulang** [reply]
我的问题和vue没有关系。但是你说的其实我也早就试过，没有用的，不是这么简单。现在的问题babel在transpile debug的时候，export这里不知道是出了什么问题，所以transpile不成功。
